### PR TITLE
Toolkit: Bump dependencies

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
This PR bumps all the toolkit dependencies either inline with Grafana core dependencies or to their latest versions.  Whilst Plugins Platform is making progress to separate toolkit into smaller packages we need to maintain toolkit and it's dependencies are either preventing plugin devs from gaining bug fixes or new features or have nested vulnerabilities that continue to show up in audits. Given we have v9 on the horizon I feel like it's a good opportunity to fix these problems which due to major dependency bumps will likely cause breaking changes.

- [x] Bump all toolkit dependencies
- [x] Migrate toolkit plugin build to use Webpack 5
- [x] Remove dependency on ts-loader
- [x] Migrate toolkit plugin linting task to use new Eslint 8 node API
- [x] ~Use browserlist to bundle plugins as ES6~ This breaks Angular plugins so have reverted in 859928e47b2d8ed3a8d9ec88ae1c75809208654f
- [x] Remove toolkit build.core task (Grafana core hasn't used it since the bump to Webpack 5)
- [x] Update toolkit README explaining how to make changes to toolkit locally
- [x] Remove pnp-webpack-plugin from input-datasource as [webpack 5 supports yarn PnP out the box](https://webpack.js.org/migrate/5/#clean-up-configuration)
- [x] Run `npx browserslist --update` to solve failing input-datasource build

Tested against the following plugins:

- [x] [clock-panel](https://github.com/grafana/clock-panel) (React, Emotion, Images)
- [x] [github-datasource](https://github.com/grafana/github-datasource) (React, Angular, Dashboards, Tests)
- [x] [grafana-gauge-panel](https://github.com/briangann/grafana-gauge-panel) (Angular, Tests)
- [x] [grafana-infinity-datasource](https://github.com/yesoreyeram/grafana-infinity-datasource) (React, Images)
- [x] [piechart-panel](https://github.com/grafana/piechart-panel) (Angular, Sass, Tests)
- [x] [pierosavi-imageit-panel](https://github.com/pierosavi/pierosavi-imageit-panel) (React, Emotion, Images)
- [x] [plugin-yarn-berry](https://github.com/jackw/plugin-yarn-berry) (React, Sass, Emotion)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

It does not bump the following dependencies to the very latest due to the latest versions being ES modules:

- ora
- globby
- execa
- chalk

### Release notice breaking change

Plugins using custom Webpack configs could potentially break due to the changes between webpack@4 and webpack@5. Please refer to the [official migration guide](https://webpack.js.org/migrate/5/) for assistance. 

Webpack 5 does not include polyfills for node.js core modules by default (e.g. `buffer`, `stream`, `os`). This can result in failed builds for plugins. If polyfills are required it is recommended to create a custom webpack config in the root of the plugin repo and add the required fallbacks:

```js
// webpack.config.js

module.exports.getWebpackConfig = (config, options) => ({
  ...config,
  resolve: {
    ...config.resolve,
    fallback: {
      os: require.resolve('os-browserify/browser'),
      stream: require.resolve('stream-browserify'),
      timers: require.resolve('timers-browserify'),
    },
  },
});
```
Please refer to the webpack build error messages or the [official migration guide](https://webpack.js.org/migrate/5/) for assistance with fallbacks.

### CODE REVIEW

File: .github/renovate.json5 [hunk 1]
```diff
@@ -31,7 +31,7 @@
     "typescript", // >4.5.x doesn't seem to work on windows with pnp currently. probably similar to https://github.com/yarnpkg/berry/issues/3722
     "ora" // we should bump this once we move to esm modules
   ],
-  "ignorePaths": ["packages/grafana-toolkit/package.json", "emails/**", "plugins-bundled/**", "**/mocks/**"],
```
Comment (gabor) [inline .github/renovate.json5:34]: are you sure about this? 😁 

i mean, isn't it better to only do this upgrade-thing before every major-grafana-release or some other, very controlled way?

Comment (jackw) [reply .github/renovate.json5:34]: Yeah you are right @gabor . I guess we want patch releases to go into toolkit deps as that would likely help reduce the number of security warnings. I'll poke around the renovate docs to see what is possible.

**Review by ashharrison90 — APPROVED** [review]
the code changes themselves look great 🙌 

full disclaimer: i haven't tested this with any plugins 😅

**Review by leventebalogh — APPROVED** [review]
Very nice work mate 👏 

Looks good to me, also tried it out locally and everything seemed to work for me. I only noticed some strange issues when tried to run it as a standalone tool (e.g. for scaffolding) locally, let's have a look at that together before merging it.

**Pull request merged**